src/MoBi.UI/Views/EditIndividualAndExpressionConfigurationsView.cs

```diff
@@ -96,14 +120,21 @@ public void SortSelectedExpressions()
          simulationExpressionsTree.Sort();
       }
 
-      private void removeSelectedExpression()
+      private void removeSelectedExpressions()
+      {
+         projectExpressionsTree.TreeView.Selection.Clear();
+         _presenter.RemoveSelectedExpressions(treeViewSelectionToTreeNodeList(simulationExpressionsTree.Selection));
+      }
+
+      private void addSelectedExpressions()
       {
-         _presenter.RemoveSelectedExpression(simulationExpressionsTree.SelectedNode);
+         simulationExpressionsTree.Selection.Clear();
+         _presenter.AddSelectedExpressions(treeViewSelectionToTreeNodeList(projectExpressionsTree.TreeView.Selection));
       }
 
-      private void addSelectedExpression()
+      private IReadOnlyList<ITreeNode> treeViewSelectionToTreeNodeList(TreeListSelection treeViewSelection)
       {
-         _presenter.AddSelectedExpression(projectExpressionsTree.TreeView.SelectedNode);
+         return treeViewSelection.All().Where(x => x.Tag is ITreeNode).Select(x => x.Tag as ITreeNode).ToList();
```

Comment:
there has to be some method to get tag as ITreeNode ion the treeview. It feels like knowing that stuff in saved as TAg etc breaks encapsulation